superset-frontend/src/dashboard/components/Dashboard.jsx

```diff
@@ -159,11 +159,27 @@ class Dashboard extends React.PureComponent {
           // added filter?
           [].push.apply(affectedChartIds, activeFilters[filterKey].scope);
         } else {
-          // changed filter field value or scope?
-          const affectedScope = (activeFilters[filterKey].scope || []).concat(
-            appliedFilters[filterKey].scope || [],
-          );
-          [].push.apply(affectedChartIds, affectedScope);
+          // has filter field value change?
+          if (
+            !areObjectsEqual(
+              appliedFilters[filterKey].values,
+              activeFilters[filterKey].values,
+            )
+          ) {
+            [].push.apply(affectedChartIds, activeFilters[filterKey].scope);
```

Comment:
NIT(optional ): I had to lookup what `[].push.apply`might do. For me `affectedChartIds.push(...activeFilters[filterKey].scope)` would be more readable (I think it's equivalent?).

Reply:
`affectedChartIds.push(...activeFilters[filterKey].scope)` just try to merge 2 arrays and assign back to `affectedChartIds`.
I agree es6 syntax is better. fixed.